campaign-launcher/interfaceV2/src/components/modals/EditApiKeyModal/index.tsx

```diff
@@ -0,0 +1,209 @@
+import { FC, useEffect } from 'react';
+
+import { yupResolver } from '@hookform/resolvers/yup';
+import {
+  Box,
+  Button,
+  FormControl,
+  FormHelperText,
+  TextField,
+  Typography,
+} from '@mui/material';
+import { Controller, useForm } from 'react-hook-form';
+import * as yup from 'yup';
+
+import { usePostExchangeApiKey } from '../../../hooks/recording-oracle';
+import FormExchangeSelect from '../../FormExchangeSelect';
+import { ModalError, ModalLoading, ModalSuccess } from '../../ModalState';
+import BaseModal from "../BaseModal";
+
+type Props = {
+  open: boolean;
+  onClose: () => void;
+  exchangeName?: string;
+  keysData?: {
+    api_key: string;
+    secret_key: string;
+  };
+};
+
+type APIKeyFormValues = {
+  apiKey: string;
+  secret: string;
+  exchange: string;
+};
+
+const validationSchema = yup.object({
+  apiKey: yup.string().required('Required'),
+  secret: yup.string().required('Required'),
+  exchange: yup.string().required('Required'),
+});
+
+const EditApiKeyModal: FC<Props> = ({ open, onClose, exchangeName, keysData }) => {
+  const { mutate: postExchangeApiKey, reset: resetMutation, isIdle, isPending, isSuccess, isError } = usePostExchangeApiKey();
+  const {
+    control,
+    formState: { errors },
+    handleSubmit,
+    reset,
+  } = useForm<APIKeyFormValues>({
+    resolver: yupResolver(validationSchema),
+    defaultValues: {
+      exchange: '',
+      apiKey: '',
+      secret: '',
+    },
+  });
+
+  useEffect(() => {
+    if (open && exchangeName && keysData) {
+      reset({
+        exchange: exchangeName || '',
+        apiKey: keysData?.api_key || '',
+        secret: keysData?.secret_key || '',
+      });
+    }
+  }, [open, exchangeName, keysData, reset]);
+
+  const handleClose = () => {
+    if (isPending) return;
+    
+    reset();
+    resetMutation();
+    onClose();
+  };
+
+  const onSubmit = (values: APIKeyFormValues) => {
+    postExchangeApiKey({
+      exchangeName: values.exchange,
+      apiKey: values.apiKey,
+      secret: values.secret,
+    });
+  };
+
+  return (
+    <BaseModal
+      open={open}
+      onClose={handleClose}
+      sx={{
+        textAlign: 'center',
+        color: 'text.primary',
+        px: { xs: 3, md: 4 },
+      }}
+    >
+      <form onSubmit={handleSubmit(onSubmit)}>
+        <Box 
+          display="flex" 
+          flexDirection="column" 
+          alignItems="center" 
+          textAlign="center" 
+          px={{ xs: 0, md: 4 }}
+        >
+          <Typography variant="h4" py={1} mb={2}>
+            Edit API key
+          </Typography>
+          {isPending && <ModalLoading />}
+          {isIdle && (
+            <>
+              <Box display="flex" gap={1} mb={3} width="100%">
+                <FormControl error={!!errors.exchange} sx={{ width: '30%' }}>
+                  <Controller
```

Comment:
I suppose it either should be disabled for `edit` form or it should be just text with exchange name, otherwise it's a bit confusing that you open edit modal for some specific exchange and then can change the select